src/components/common/DelayedRender.test.ts

```diff
@@ -0,0 +1,26 @@
+import { act, render, screen } from '@testing-library/react'
+import { div, h } from 'react-hyperscript-helpers'
+
+import { DelayedRender } from './DelayedRender'
+
+
+describe('DelayedRender', () => {
+  it('renders children after a delay', () => {
+    // Arrange
+    jest.useFakeTimers()
+
+    // Act
+    render(h(DelayedRender, { delay: 3000 }, [
+      div(['Hello world'])
+    ]))
+
+    // Assert
+    expect(screen.queryByText('Hello world')).toBeNull()
```

Comment:
this and other intermediate assertions can be switched to saved values to be asserted later in a unified // Assert section.

```
const initialIsContentRendered = screen.queryByText('Hello world') === null
// ...
// Assert
expect(initialIsContentRendered).toBe(false)
```

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/3795/commits/39369d085bfd9bca37e0bfb661f8757a85cb00f9. I still find switching between act and assert more convenient and sometimes easier to read than storing intermediate values to keep all assertions at the end.